File: src/Domain/Entities/Games/Round.cs
using Domain.Enums;

namespace Domain.Entities.Games;

public class Round
{
    public int Number { get; set; }
    public RoundStatus Status { get; set; }
    public string GameMode { get; set; }
    public List<string> RestrictedLanguages { get; set; }
    public string ChallengeSelectorType { get; set; }
    public DateTime? StartTime { get; set; }
    public int DurationMinutes { get; set; }
    public string ChallengeId { get; set; }
    public List<RoundSummary> RoundSummaries { get; set; }
}

File: src/react-frontend/src/app/types/models/game/round.ts
import { ChallengeSelectorType } from 'app/types/enums/challengeSelectorType';
import { GameMode } from 'app/types/enums/gameMode';
import { RoundStatus } from 'app/types/enums/roundStatus';
import { Challenge } from '../challenge/challenge';
import { RoundSummary } from './roundSummary';

export type Round = {
  number: number;
  status: RoundStatus;
  gameMode: GameMode;
  restrictedLanguages: string[];
  challengeSelectorType: ChallengeSelectorType;
  startTime?: Date;
  durationMinutes: number;
  challenge: Challenge;
  roundSummaries: RoundSummary[];
};
